plugins/ptp_operator/metrics/metrics.go

```diff
@@ -0,0 +1,420 @@
+	locked                      = "s2"
+	maxOffsetThrushHold float64 = 100
+	minOffsetThrushHold float64 = -100
+)
+
+var (
+	// NodeName from the env
+	ptpNodeName = ""
+	eventConfig *common.SCConfiguration
+
+	// OffsetFromMaster metrics for offset from the master
+	OffsetFromMaster = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ptpNamespace,
+			Subsystem: ptpSubsystem,
+			Name:      "offset_from_master",
+			Help:      "",
+		}, []string{"process", "node", "iface"})
+
+	// MaxOffsetFromMaster  metrics for max offset from the master
+	MaxOffsetFromMaster = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ptpNamespace,
+			Subsystem: ptpSubsystem,
+			Name:      "max_offset_from_master",
+			Help:      "",
+		}, []string{"process", "node", "iface"})
+
+	// FrequencyAdjustment metrics to show frequency adjustment
+	FrequencyAdjustment = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ptpNamespace,
+			Subsystem: ptpSubsystem,
+			Name:      "frequency_adjustment",
+			Help:      "",
+		}, []string{"process", "node", "iface"})
+
+	// DelayFromMaster metrics to show delay from the master
+	DelayFromMaster = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ptpNamespace,
+			Subsystem: ptpSubsystem,
+			Name:      "delay_from_master",
+			Help:      "",
+		}, []string{"process", "node", "iface"})
+)
+
+var registerMetrics sync.Once
+
+// RegisterMetrics ... register metrics for all side car plugins
+func RegisterMetrics(nodeName string) {
+	registerMetrics.Do(func() {
+		prometheus.MustRegister(OffsetFromMaster)
+		prometheus.MustRegister(MaxOffsetFromMaster)
+		prometheus.MustRegister(FrequencyAdjustment)
+		prometheus.MustRegister(DelayFromMaster)
+
+		// Including these stats kills performance when Prometheus polls with multiple targets
+		prometheus.Unregister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+		prometheus.Unregister(prometheus.NewGoCollector())
+
+		ptpNodeName = nodeName
+	})
+}
+
+// Stats calculates stats  nolint:unused
+type Stats struct {
+	num                 float64
+	max                 float64
+	min                 float64
+	mean                float64
+	sumSqr              float64
+	sumDiffSqr          float64
+	frequencyAdjustment float64
+	delayFromMaster     float64
+	lastOffset          float64 //nolint:structcheck
+	lastClockState      ceevent.SyncState
+}
+
+func (s *Stats) addValue(val float64) {
+
+	oldMean := s.mean
+
+	if s.max < val {
+		s.max = val
+	}
+	if s.min > val {
+		s.min = val
+	}
+	s.num++
+	s.mean = oldMean + (val-oldMean)/s.num
+	s.sumSqr += val * val
+	s.sumDiffSqr += (val - oldMean) * (val - s.mean)
+
+}
+
+// get stdDev
+func (s *Stats) getStdev() float64 { //nolint:unused
+	if s.num > 0 {
+		return math.Sqrt(s.sumDiffSqr / s.num)
+	}
+	return 1
+}
+func (s *Stats) getMaxAbs() float64 {
+	if s.max > s.mean {
+		return s.max
+	}
+	return s.min
+
+}
+
+// reset status
+func (s *Stats) reset() { //nolint:unused
```

Comment:
Does `s.min` need to be reset?

Reply:
Added and also  making reset call when its not in Locked state as per PTP4l log